cleanlab/rank.py

```diff
@@ -152,14 +164,320 @@ def get_normalized_margin_for_each_label(
 
     Returns
     -------
-      normalized_margin : np.array (float)
-        Return a score (between 0 and 1) for each example of its likelihood of
-        being correctly labeled. Assumes pred_probs is computed holdout/out-of-sample.
-        normalized_margin = prob_label - max_prob_not_label"""
+    normalized_margin : np.array (float)
+      Return a score (between 0 and 1) for each example of its likelihood of
+      being correctly labeled. Assumes pred_probs is computed holdout/out-of-sample.
+      normalized_margin = prob_label - max_prob_not_label"""
 
     self_confidence = get_self_confidence_for_each_label(labels, pred_probs)
     max_prob_not_label = np.array(
         [max(np.delete(pred_probs[i], l, -1)) for i, l in enumerate(labels)]
     )
-    label_quality_scores = (self_confidence - max_prob_not_label + 1) / 2
+    normalized_margin = (self_confidence - max_prob_not_label + 1) / 2
+    return normalized_margin
+
+
+def get_confidence_weighted_entropy_for_each_label(
+    labels: np.array, pred_probs: np.array
+) -> np.array:
+    """Returns the normalized entropy divided by "self-confidence".
+
+    Score is between 0 and 1.
+    1 - clean label (not an error).
+    0 - dirty label (label error).
```

Comment:
This should live in the `score_label_quality()` docstring as behavior that applies to all scoring functions. So it should not be in the docstring for individual scoring functions.  In the individual scoring functions' docstrings, I'd just put something simpler like:

"This is a function to compute label-quality scores, where lower scores indicate labels less likely to be correct."

Reply:
I'd also change the language to:
```
 1 - clean label (given label is likely correct).
 0 - dirty label (given label is likely incorrect).
```

Reply:
Agree, sounds good. Done: 100783d9199110a66737f2d09e2fce762b500c45